File: java/test/com/nrkei/microservices/rapids_rivers/PacketProblemsTest.java
package com.nrkei.microservices.rapids_rivers;

import org.junit.Test;

import static org.junit.Assert.*;

/*
 * Copyright (c) 2016 by Fred George
 * May be used freely except for training; license required for training.
 * @author Fred George
 */

// Ensures that PacketProblems operates correctly
public class PacketProblemsTest {

    private final static String VALID_JSON =
            "{\"key1\":\"value1\"}";

    @Test
    public void noProblemsFoundDefault() throws Exception {
        assertFalse(new PacketProblems(VALID_JSON).hasErrors());
    }
}

File: csharp_v2_beta/test/MicroServiceWorkshop.Tests/PacketProblemsTest.cs
/* 
 * Copyright (c) 2016 by Fred George
 * May be used freely except for training; license required for training.
 */

using NUnit.Framework;

namespace MicroServiceWorkshop.Tests
{
    // Ensures that PacketProblems operates correctly
    [TestFixture]
    public class PacketProblemsTest
    {
        private static readonly string ValidJson = "{\"key1\":\"value1\"}";
        private PacketProblems _problems;

        [SetUp]
        public void SetUp()
        {
            _problems = new PacketProblems(ValidJson);
        }

        [Test]
        public void NoProblemsFoundDefault()
        {
            Assert.False(_problems.HasProblems());
        }
    }
}